javascript_ts/11.qnamaker/src/myAppInsightsQnAMaker.ts

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { TelemetryClient } from 'applicationinsights';
+import { TurnContext } from 'botbuilder';
+import { QnAMaker, QnAMakerEndpoint, QnAMakerOptions, QnAMakerResult } from 'botbuilder-ai';
+import { EventTelemetry } from 'applicationinsights/out/Declarations/Contracts';
+
+/**
+ * Custom wrapper around QnAMaker from botbuilder-ai.
+ * Sends custom events to Application Insights.
+ */
+export class MyAppInsightsQnAMaker extends QnAMaker {
+    
+    public readonly QnAMsgEvent = 'QnAMessage';
+
+    public requestConfig: QnAMakerOptions;
+    public logOriginalMessage: boolean = false;
+    public logUserName: boolean = false;
+
+    constructor(endpoint: QnAMakerEndpoint, options?: QnAMakerOptions, logOriginalMessage?: boolean, logUserName?: boolean) {
+        super(endpoint);
+        this.requestConfig = options;
+        this.logOriginalMessage = logOriginalMessage;
+        this.logUserName = logUserName;
+    }
+
+    /**
+     * Calls QnA Maker and then sends a custom Event to Application Insights with the answer and question that matched closest with the user's utterance.
```

Comment:
>answer and quest [](start = 86, length = 16)

question and answer, rather than answer and question. :)

Reply:
Removed AppInsights Middleware. #Resolved